command/agent/local.go

```diff
@@ -109,18 +109,18 @@ func (l *localState) ConsulServerUp() {
 // Pause is used to pause state synchronization, this can be
 // used to make batch changes
 func (l *localState) Pause() {
-	atomic.StoreInt32(&l.paused, 1)
+	atomic.AddInt32(&l.paused, 1)
 }
 
 // Resume is used to resume state synchronization
 func (l *localState) Resume() {
-	atomic.StoreInt32(&l.paused, 0)
+	atomic.AddInt32(&l.paused, -1)
```

Comment:
Since this returns the new value, I think I would put the panic in here to immediately catch any mismatch of these.